src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -403,199 +416,65 @@ object ApiOperations {
-              }
-
-            case Some(x) if x == "fitness" =>
-              order.map(_.toLowerCase) match {
-                case Some(x) if x == "asc" => BlocksAction(blockAction.sortBy(_._6.asc))
-                case Some(x) if x == "desc" => BlocksAction(blockAction.sortBy(_._6.desc))
-                case None => BlocksAction(blockAction.sortBy(_._6.desc))
-              }
-            case Some(x) if x == "context" =>
-              order.map(_.toLowerCase) match {
-                case Some(x) if x == "asc" => BlocksAction(blockAction.sortBy(_._7.asc))
-                case Some(x) if x == "desc" => BlocksAction(blockAction.sortBy(_._7.desc))
-                case None => BlocksAction(blockAction.sortBy(_._7.desc))
-              }
-            case Some(x) if x == "signature" =>
-              order.map(_.toLowerCase) match {
-                case Some(x) if x == "asc" => BlocksAction(blockAction.sortBy(_._8.asc))
-                case Some(x) if x == "desc" => BlocksAction(blockAction.sortBy(_._8.desc))
-                case None => BlocksAction(blockAction.sortBy(_._8.desc))
-              }
-            case Some(x) if x == "protocol" =>
-              order.map(_.toLowerCase) match {
-                case Some(x) if x == "asc" => BlocksAction(blockAction.sortBy(_._9.asc))
-                case Some(x) if x == "desc" => BlocksAction(blockAction.sortBy(_._9.desc))
-                case None => BlocksAction(blockAction.sortBy(_._9.desc))
-              }
-            case Some(x) if x == "chain_id" =>
-              order.map(_.toLowerCase) match {
-                case Some(x) if x == "asc" => BlocksAction(blockAction.sortBy(_._10.asc))
-                case Some(x) if x == "desc" => BlocksAction(blockAction.sortBy(_._10.desc))
-                case None => BlocksAction(blockAction.sortBy(_._10.desc))
-              }
-            case Some(x) if x == "hash" =>
-              order.map(_.toLowerCase) match {
-                case Some(x) if x == "asc" => BlocksAction(blockAction.sortBy(_._11.asc))
-                case Some(x) if x == "desc" => BlocksAction(blockAction.sortBy(_._11.desc))
-                case None => BlocksAction(blockAction.sortBy(_._11.desc))
-              }
-            case Some(x) if x == "operations_hash" =>
-              order.map(_.toLowerCase) match {
-                case Some(x) if x == "asc" => BlocksAction(blockAction.sortBy(_._12.asc))
-                case Some(x) if x == "desc" => BlocksAction(blockAction.sortBy(_._12.desc))
-                case None => BlocksAction(blockAction.sortBy(_._12.desc))
-              }
-            case None =>
-              order.map(_.toLowerCase) match {
-                case Some(x) if x == "asc" => BlocksAction(blockAction.sortBy(_._1.asc))
-                case Some(x) if x == "desc" => BlocksAction(blockAction.sortBy(_._1.desc))
-                case None => BlocksAction(blockAction.sortBy(_._1.desc))
-              }
+        case blocks: BlocksAction =>
```

Comment:
Should we eventually create ADTs for schema column names, or do you think string pattern matching is fine for this use case?

Reply:
Right now I wouldn't go so far as to define a type for each column.

It will turn useful to use a "newtype wrapper" for the `String` to distinguish between sorting-field names and sorting-order for example... maybe you meant that?